plugin/forward/proxy.go

```diff
@@ -7,78 +7,56 @@ import (
 	"time"
 
 	"github.com/coredns/coredns/plugin/pkg/up"
-
-	"github.com/miekg/dns"
 )
 
 // Proxy defines an upstream host.
 type Proxy struct {
 	avgRtt int64
 	fails  uint32
 
-	addr   string
-	client *dns.Client
+	addr string
 
 	// Connection caching
 	expire    time.Duration
 	transport *transport
 
 	// health checking
-	probe *up.Probe
+	probe  *up.Probe
+	health HealthChecker
 }
 
 // NewProxy returns a new proxy.
-func NewProxy(addr string, tlsConfig *tls.Config) *Proxy {
+func NewProxy(addr string, protocol int, cfg *tls.Config) *Proxy {
```

Comment:
consider removing the `cfg` parameter - we never pass tls config to `NewProxy()`. The same for second parameter of `newTransport()`. The tls config is actually set with `SetTlsConfig()` method.